### PR TITLE
Refactor facebook

### PR DESCRIPTION
**Pull request by BlueHotDog**

### CODE REVIEW

File: SeaSick/User.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//
+//
+
+#import <Foundation/Foundation.h>
+#import <CoreData/CoreData.h>
+
+
+@interface User : NSManagedObject
+
+@property (nonatomic, retain) NSString * facebookEmail;
```
Comment (itayadler) [inline SeaSick/User.h:15]: remove

File: SeaSick/Models/SCKUser+CurrentUser.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#import "SCKUser.h"
+#import "SCKAuthentication.h"
+
+@interface SCKUser (CurrentUser)
+
+@property (nonatomic, retain, readonly) NSString *email;
+@property (nonatomic, retain, readonly) NSString *name;
+
++ (SCKUser *)currentUser;
++ (SCKUser *)createFromAuthentication:(SCKAuthentication *)authentication;
```
Comment (itayadler) [inline SeaSick/Models/SCKUser+CurrentUser.h:19]: perhaps call it findOrCreate

File: SeaSick/Models/SCKUser+CurrentUser.m [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+#import "SCKFacebookAuthentication+Login.h"
+
+@implementation SCKUser (CurrentUser)
+
+
+static SCKUser *_currentUser;
+
+- (NSString *)email
+{
+    return ((SCKAuthentication *)[[self authentications] anyObject]).email;
```
Comment (itayadler) [inline SeaSick/Models/SCKUser+CurrentUser.m:21]: anyObjeet? :x

File: SeaSick/User.h (hunk 1 above)
Comment (ShayDavidson) [inline SeaSick/User.h:15]: as a rule of thumb, when you expose NSString, NSArray and NSDictionary in their immutable forms, add `copy`.

edit: whoopse, it's auto generated, but keep that in mind for classes you write youself :)

File: SeaSick/Models/SCKUser+CurrentUser.m [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+{
+    SCKUser *user = [SCKUser MR_createEntity];
+    [user addAuthenticationsObject:authentication];
+    [[NSManagedObjectContext MR_defaultContext] MR_saveToPersistentStoreAndWait];
+    return user;
+}
+
+- (void)logout
+{
+//    [[self authentications] enumerateObjectsUsingBlock:^(id obj, BOOL *stop) {
```
Comment (giladgo) [inline SeaSick/Models/SCKUser+CurrentUser.m:51]: Why is this commented out?

File: SeaSick/Models/SCKUser+CurrentUser.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+//
+//
+
+#import <Foundation/Foundation.h>
+#import "SCKUser.h"
+#import "SCKAuthentication.h"
+
+@interface SCKUser (CurrentUser)
+
+@property (nonatomic, retain, readonly) NSString *email;
```
Comment (ShayDavidson) [inline SeaSick/Models/SCKUser+CurrentUser.h:15]: look at what i wrote below for immutable properties.

File: SeaSick/Models/SCKUser+CurrentUser.m (hunk 3 above)
Comment (giladgo) [inline SeaSick/Models/SCKUser+CurrentUser.m:21]: sets don't have an order so basically first==any

**Pull request closed without merging**